File: TypeScript/leetcode/sort-integers-by-the-number-of-1-bits/solution.ts
function sortByBits(arr: number[]): number[] {
    const bits = (x: number): number => { let c = 0; while (x > 0) { c += x & 1; x >>>= 1; } return c; };
    return arr.slice().sort((a, b) => bits(a) - bits(b) || a - b);
}

File: Go/leetcode/sort-integers-by-the-number-of-1-bits/solution.go
import (
    "math/bits"
    "sort"
)

func sortByBits(arr []int) []int {
    sort.Slice(arr, func(i, j int) bool {
        bi, bj := bits.OnesCount(uint(arr[i])), bits.OnesCount(uint(arr[j]))
        if bi != bj {
            return bi < bj
        }
        return arr[i] < arr[j]
    })
    return arr
}
